### ISSUE TITLE
ispindel tilting issue

### ISSUE THREAD

**Issue by ghost** [universam1/iSpindel#340]
A recap of what I did with my ispindel:
1. soldered, assembled and connected: also verified the gyroscope was working as well as the temperature sensor;
2. put weights on until had a tilt of 24;
3. offset calibration as per ispindel menu "maintenance" where it says to do the offset before definitive calibration and result was a tilt of 89.80;
4. placed the ispindel back in plain water and the tilt was 60 whereas in the same position I had 24... With the apparatus in upright position it gives a tilt of 62...
Is that normal?
Do I have to start the calibration with sugar even if my starting tilt is 60?
Thanks to anyone that could help

**Comment by ErikdBr** [comment]
You should have done point 3 before point 2, so first the offset calibration and after that put weights on until... in plain water.

**Comment by ftassini** [comment]
Thanks Erik, so what should I do now?
Maybe reset the ispindel or re-flash the firmware?
What would you suggest?

**Comment by ErikdBr** [comment]
Initializing the iSpindel(see the FAQ at What should I do if my iSpindel isn't working properly after flashing it with the firmware?) and flash the software again, is allways a good idea if you have any problems I would do the calibration again, first the offset calibration and after that add or remove weights to get a tilt of around 20.

**Comment by ftassini** [comment]
Hi Erik,
I've reflashed the ispindel and I've done the calibration offset BEFORE putting the weights on, but I still have a minimum tilt of circa 60...
Any idea?

**Comment by ErikdBr** [comment]
What was the angle in plain water after putting the weights on? And do you have some picture of your ispindel?

**Comment by ftassini** [comment]
With angle you mean approx. degrees?
Wouldn't know exactly, I'll take a picture and upload it asap

![IMG_20200129_094505](https://user-images.githubusercontent.com/60745743/74351387-56de8900-4db7-11ea-9fd4-524f23174b57.jpg)
![IMG_20200129_094621](https://user-images.githubusercontent.com/60745743/74351391-580fb600-4db7-11ea-99fe-125c130348d8.jpg)
![IMG_20200129_094629_BURST001_COVER](https://user-images.githubusercontent.com/60745743/74351393-5940e300-4db7-11ea-8439-fda8af325426.jpg)
![IMG_20200129_094629_BURST002](https://user-images.githubusercontent.com/60745743/74351396-59d97980-4db7-11ea-9092-2b990354f2b0.jpg)
![IMG_20200129_094828](https://user-images.githubusercontent.com/60745743/74351397-5a721000-4db7-11ea-81e1-d1d1dd9cdcde.jpg)
![IMG_20200212_163735](https://user-images.githubusercontent.com/60745743/74351424-63fb7800-4db7-11ea-8f4e-9f21b7f5fa96.jpg)
![IMG_20200212_163745](https://user-images.githubusercontent.com/60745743/74351430-65c53b80-4db7-11ea-93f3-19db1d356d2b.jpg)
![IMG_20200212_163757](https://user-images.githubusercontent.com/60745743/74351432-665dd200-4db7-11ea-9be3-7c1ba0880292.jpg)
![IMG_20200212_163823](https://user-images.githubusercontent.com/60745743/74351435-678eff00-4db7-11ea-8177-f1df1a573de4.jpg)

**Comment by ftassini** [comment]
Position and tilt result with bolt and 2 nuts on
![IMG_20200212_165655](https://user-images.githubusercontent.com/60745743/74352638-1718a100-4db9-11ea-9747-d3b60de31900.jpg)
![IMG_20200212_165710](https://user-images.githubusercontent.com/60745743/74352645-18e26480-4db9-11ea-842c-a8f12f576d3d.jpg)

**Comment by ErikdBr** [comment]
I assume and think I see it in the last picture the battery side is downwards facing the bottom of the pan. Did you do the offset calibration with the same side, the battery side, facing down to the flat and level surface? If you did I am lost, if you didn't then that could be the problem.

**Comment by ftassini** [comment]
I actually did the calibration in the way you said...

**Comment by ErikdBr** [comment]
@universam1 I am completely lost here. Do you have any idea?

**Comment by ecsfang** [comment]
Hi,

For me it looks as if the accelerometer is mounted upside-down. Looking at the picture, it looks as if the components on the MPU-6050 are towards the pbc, and the Vcc pin is closest to the bottom.
But looking at the pictures on https://3d-mechatronics.de/en/ispindel-teilmontiert/-partially-assembled_962.html, it looks at if the components should be facing outwards, and Vcc closest to the lid.
Just my two cents - I don't have this board myself, so I'm just relying on the pictures ...
![image](https://user-images.githubusercontent.com/8043784/74408148-efb2ea00-4e33-11ea-8ac2-e51bed96b899.png)

**Comment by ftassini** [comment]
@ecsfang You're right, the picture you refer to is clear, but I soldered the MPU according with the "DIY iSpindel manual" which shows the opposite, on page 9, pictures 17 and 18....

**Comment by ecsfang** [comment]
@ftassini Agree, looking at https://dl.dropbox.com/s/0a2jkkvf0ba9g0c/DIY_Spindel_Anleitung_PCB1_EN_compressed.pdf it shows just as you have mounted the module, so I guess that that is the correct way!
I was just fooled by the picture above ... ;)

**Comment by ftassini** [comment]
@universam1 could you give me your opinion on this issue please?

**Comment by ErikdBr** [comment]
Only thing I can add is that in the older design the accelerometer is facing the other way around(with the chips up) as can be seen in the pictures. Also I can see a emailadres from 3d-mechatronics where you can ask questions, maybe you can ask which side should face up?
![20190512_114808](https://user-images.githubusercontent.com/48140503/74588033-81b81f80-4ff9-11ea-8af1-3b9ab0b5213e.jpg)
![20190512_114826](https://user-images.githubusercontent.com/48140503/74588035-85e43d00-4ff9-11ea-9ab9-86b3544a188e.jpg)

**Comment by majberger** [comment]
> 
> 
> @ftassini Agree, looking at https://dl.dropbox.com/s/0a2jkkvf0ba9g0c/DIY_Spindel_Anleitung_PCB1_EN_compressed.pdf it shows just as you have mounted the module, so I guess that that is the correct way!
> I was just fooled by the picture above ... ;)

I've just finished soldering two brand new iSpindel. Following the copper on the PCB you can see that VCC and GND is correct the way it's documented and the way ftassini have built it.

**Comment by ErikdBr** [comment]
> > @ftassini Agree, looking at https://dl.dropbox.com/s/0a2jkkvf0ba9g0c/DIY_Spindel_Anleitung_PCB1_EN_compressed.pdf it shows just as you have mounted the module, so I guess that that is the correct way!
> > I was just fooled by the picture above ... ;)
> 
> I've just finished soldering two brand new iSpindel. Following the copper on the PCB you can see that VCC and GND is correct the way it's documented and the way ftassini have built it.

And yours are working correctly? Cause if they do maybe you can help.

**Comment by ftassini** [comment]
@ErikdBr as I wrote @ecsfang I just followed instructions given by DIY_Spindel_Anleitung_PCB1_EN_compressed.pdf ....
I've already tried asking at 3d mechatronics but the guy Tjorben Eberle told me he's only responsible for hardware and for what he sees it is working; he sent me over here to open an issue...

**Comment by majberger** [comment]
> 
> 
> > > @ftassini Agree, looking at https://dl.dropbox.com/s/0a2jkkvf0ba9g0c/DIY_Spindel_Anleitung_PCB1_EN_compressed.pdf it shows just as you have mounted the module, so I guess that that is the correct way!
> > > I was just fooled by the picture above ... ;)
> > 
> > 
> > I've just finished soldering two brand new iSpindel. Following the copper on the PCB you can see that VCC and GND is correct the way it's documented and the way ftassini have built it.
> 
> And yours are working correctly? Cause if they do maybe you can help.

Will report back when I've had time to try them out. Haven't even flashed them yet, way too many other things that need my attention right now. Just wanted to report that the manual seems to be correct when "reading" the PCB.

**Comment by Fatfil** [comment]
I have a different kit, from Colin on ebay, but have exactly the same issue. Did you fix this ghost??

**Comment by Fatfil** [comment]
Fixed it. Make sure the gyro is facing upwards when running the offset calibration routine even if when you float it, it faces down. If you do this the problem goes away. I would recommend you play with the bouncy so that the iSpindel floats with the gyro pointing upwards.

**Comment by whitetd** [comment]
had to fix mine, flipped the gyro board over, extended the wires, I will add a pic when I get a chance.

**Comment by majberger** [comment]
Finally sold the house and have finished moving, sorry for not getting back to you guys earlier.

TL;DR - I have the exact same model that ftassini (account deleted, hence the name "ghost") was using. Can reproduce the error and confirm that Fatfil's solution (make sure the gyro face "upwards" during calibration) works. You can close this issue.


I was a bit at a loss first.
Had to get one of my iSpindels ready in a hurry, since I had promised my father to brew him a Czech Premium Lager for summer. It worked really well! The second one I popped into a fermenter last sunday, and it had the exact same issue as ftassini's had. Really weird!

Bottled my fathers Lager a few days ago and now I have tested with the one that worked. I simply had forgotten to use Maintenance->calibrate. As soon as I'd done that it too had the same issue. Using the serial debug output I could see that it calibrated to "new offsets: -966:-1020:5224"

Suspected that the calibration optimization was to blame, flashed it with older firmware with the old "this takes several minutes"-calibration. Same odd result.

Set up so that I could dig into the code, compile my own firmware and really start debugging. Then, on a hunch, I checked the comments and found Fatfil's solution. Many thanks for posting that, it probably saved me lots of time! So I reset the Wemos D1 and flashed the latest firmware, made sure that it was "upside down" so that the gyro faced upwards and calibrated once more: "new offsets: -988:-1156:1628". 
Hooray! Now vertical is ~90 degrees and horizontal is ~0 degrees.

If you want to implement a software solution when calibrating (if weird readings, assume it's an upside-down-kit and calibrate correctly) I guess it would be nice. But I opt for just mentioning the requirement of having the gyro facing upwards in the calibration documentation. Better spend the time coding other things. :-)

**Comment by universam1** [comment]
Thank you @majberger  for that detailed feedback!
Would you like to add a PR with your findings to the FAQ, might be helpful for other in similar situations

**Comment by universam1** [comment]
Try using the latest firmware version and report if that doesn’t improve the situation
